pallets/lbp/src/lib.rs

```diff
@@ -667,15 +679,18 @@ impl<T: Config> Pallet<T> {
 		let now = T::BlockNumberProvider::current_block_number();
 
 		ensure!(
-			(pool_data.start.is_zero() && pool_data.end.is_zero())
-				|| (now < pool_data.start && pool_data.start < pool_data.end),
+			match (pool_data.start, pool_data.end) {
+				(Some(start), Some(end)) => now < start && start < end,
+				(None, None) => true,
+				_ => false
+			},
 			Error::<T>::InvalidBlockRange
 		);
 
 		// this restriction is based on the AtLeast32Bit trait of the frame_system::Balance type
 		// and is expected by the calculate_linear_weights function
 		ensure!(
-			pool_data.end.saturating_sub(pool_data.start) <= u32::MAX.into(),
+			pool_data.end.unwrap_or_default().saturating_sub(pool_data.start.unwrap_or_default()) <= u32::MAX.into(),
 			Error::<T>::MaxSaleDurationExceeded
```

Comment:
We might want to constraint this to days since this blocks creation of any other LBP with the same assets and is storing data on-chain. We'd want to transform to amm after max ~ month?

Reply:
sure, we can have that, this is more of technical restriction

Reply:
Perhaps I am missing something - but what does this have to do with Balance type ? 

This check will be always true because the start and end are u32 types too.

Reply:
There is probably a typo. It should be `BlockNumber` type. Yes, it is `u32` right now, but other types are also valid.

Reply:
[That's](https://github.com/galacticcouncil/Basilisk-node/blob/17f54ed443a54adc05eab44b740507db2995b253/pallets/price-oracle/src/lib.rs#L259) how we set the weight.
You can see [here](https://github.com/galacticcouncil/Basilisk-node/blob/17f54ed443a54adc05eab44b740507db2995b253/pallets/xyk/src/lib.rs#L528) how it is used in the xyk pallet

Reply:
fixed by #253